express-delete/server.ts

```diff
@@ -35,11 +35,23 @@ app.get('/api/grades', (req, res) => {
 });
 
 app.delete('/api/grades/:id', (req, res) => {
-  for (const key in grades) {
-    if (Number(req.params.id) === grades[key].id) {
-      delete grades[key];
-      res.sendStatus(204);
+  const id = Number(req.params.id);
+  try {
+    for (const key in grades) {
+      if (id === grades[key].id) {
+        delete grades[key];
+        res.sendStatus(204);
+        break;
+      }
+      if (id !== grades[key].id) {
+        res.sendStatus(404);
+        throw new Error(
```

Comment:
You do not want to do this. This will be caught in the catch block (good) and log the error (bad). You don't want to log the error because it's a user error, not a server error. If you log every user error you will end up logging all kinds of stuff you can't do anything about. In a server, you only want to log errors that identify potential bugs in your code.